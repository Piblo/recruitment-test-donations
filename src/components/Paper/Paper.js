import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Paper = ({ children, className }) => (
  <Container className={className}>{children}</Container>
);

export default Paper;

Paper.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};


const Container = styled.div`
  background-color: #ffffff;
  padding: 16px 25px;
  position: relative;
  box-shadow: 0 2px 4px 0 rgba(194, 194, 194, 0.5);
`;
