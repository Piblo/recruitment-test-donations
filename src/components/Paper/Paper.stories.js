import React from 'react';
import { storiesOf } from '@storybook/react';
import Paper from './Paper';

storiesOf('Paper', module)
  .add('with children', () => (
    <Paper>
      <h2>This is a paper component</h2>
      <p>It renders anything you pass to it as children and applies styles consistently</p>
    </Paper>
  ));
