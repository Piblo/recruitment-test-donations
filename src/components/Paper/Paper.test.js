import React from 'react';
import { shallow } from 'enzyme';
import Paper from './Paper';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('<Paper>', () => {
  it('Renders childen prop', () => {
    const children = <div>Child element</div>;
    const component = shallow(<Paper>{children}</Paper>);
    expect(component.children().getElement()).toEqual(children);
  });

  it('Matches the snapshot', () => {
    const tree = renderer.create(<Paper>children</Paper>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
