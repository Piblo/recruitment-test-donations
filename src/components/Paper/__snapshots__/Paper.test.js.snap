// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paper> Matches the snapshot 1`] = `
.c0 {
  background-color: #ffffff;
  padding: 16px 25px;
  position: relative;
  box-shadow: 0 2px 4px 0 rgba(194,194,194,0.5);
}

<div
  className="c0"
>
  children
</div>
`;
